versioned/spi/src/main/java/org/projectnessie/versioned/RefLogEntry.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned;
+
+import java.util.List;
+import org.immutables.value.Value;
+
+/** Represents a reflog-entry stored in the database. */
+@Value.Immutable
+public interface RefLogEntry {
+
+  Hash getRefLogId();
+
+  String getRefName();
+
+  String getRefType();
+
+  Hash getCommitHash();
+
+  /** Zero, one or more ancestors of this entry, nearest parent first. */
+  List<Ancestor> getAncestors();
```

Comment:
finally I need list of commit hashes for gc to avoid fetching each row. But I want last reflog id also to continue calling back the api again. So, storing both parent reflog and commit hash

Reply:
First, it's still an implementation detail - second, you have the commit hash in RefLogEntry in the RefLogResponse, no?